components/rhs_thread/rhs_thread.test.jsx

```diff
@@ -48,26 +48,32 @@ describe('components/RhsThread', () => {
 
     test('should match snapshot', () => {
         const wrapper = shallow(
-            <RhsThread {...baseProps}/>
+            <RhsThread {...baseProps}/>,
+            {disableLifecycleMethods: true},
         );
         expect(wrapper).toMatchSnapshot();
     });
 
     test('should make api call to get thread posts on socket reconnect', () => {
         const wrapper = shallow(
-            <RhsThread {...baseProps}/>
+            <RhsThread {...baseProps}/>,
+            {disableLifecycleMethods: true},
         );
-
-        wrapper.setProps({socketConnectionStatus: false});
+        const prevProps = {
+            ...baseProps,
+            socketConnectionStatus: false,
+        };
         wrapper.setProps({socketConnectionStatus: true});
+        wrapper.instance().componentDidUpdate(prevProps);
```

Comment:
To test the function within the lifecycle method, I'm manually calling `componentDidUpdate` and passing in a mock `prevProps` object.